File: packages/gbnf-experiment/data/20260909T140517Z_670261ce/ported_implementation/utils/errors/get_input_as_string.py
"""Port of ``src/utils/errors/get-input-as-string.ts``."""

from __future__ import annotations

from typing import List, Sequence, Union

ValidInput = Union[str, int, Sequence[int]]


def get_input_as_string(src: ValidInput) -> str:
    if isinstance(src, str):
        return src

    code_points: List[int] = list(src) if isinstance(src, (list, tuple)) else [src]
    return "".join(chr(cp) for cp in code_points)


getInputAsString = get_input_as_string

File: packages/gbnf-experiment/data/20260909T150124Z_a7cd2ee9/ported_implementation/src/utils/errors/get-input-as-string.ts
import type { ValidInput } from './errors-types';

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  if (typeof src === 'number') {
    return String.fromCodePoint(src);
  }
  return src.map((cp) => String.fromCodePoint(cp)).join('');
};
